Main.py

```diff
@@ -200,14 +200,18 @@ def find_common_pool(players: Set[int], shared_pool: Set[str]):
         itemcount = len(world.itempool)
         world.itempool = new_itempool
 
-        # can produce more items than were removed
         while itemcount > len(world.itempool):
+            items_to_add = []
             for player in group["players"]:
                 if group["replacement_items"][player]:
-                    world.itempool.append(AutoWorld.call_single(world, "create_item", player,
+                    items_to_add.append(AutoWorld.call_single(world, "create_item", player,
                                                                 group["replacement_items"][player]))
                 else:
-                    AutoWorld.call_single(world, "create_filler", player)
+                    items_to_add.append(AutoWorld.call_single(world, "create_filler", player))
+            world.random.shuffle(items_to_add)
+            while itemcount > len(world.itempool) and len(items_to_add) > 0:
```

Comment:
instead of doing individual list pop and appends, a single .extend(items[wanted:]) should execute faster